lib/checkunusedvar.cpp

```diff
@@ -1489,8 +1494,35 @@ bool CheckUnusedVar::isRecordTypeWithoutSideEffects(const Type* type)
         if (f.argDef && Token::simpleMatch(f.argDef->link(), ") ="))
             continue; // ignore default/deleted constructors
         const bool emptyBody = (f.functionScope && Token::simpleMatch(f.functionScope->bodyStart, "{ }"));
-        const bool hasInitList = f.argDef && Token::simpleMatch(f.argDef->link(), ") :");
-        if (!emptyBody || hasInitList)
+
+        Token* nextToken = f.argDef->link();
+        if ( (nextToken->str() == ")") && (nextToken->next()->str() == ":") ) {
```

Comment:
I would prefer `Token::simpleMatch` here. It's shorter code and for the "compiled pattern match" there is no significant speed penalty.